File: lib/domain/field/results/field_with_farm.rb
# frozen_string_literal: true

module Domain
  module Field
    module Results
      # 圃場詳細ユースケースの戻り: 認可済み農場コンテキスト付き（表現形式に依存しない）。
      class FieldWithFarm
        attr_reader :farm, :field

        def initialize(farm:, field:)
          @farm = farm
          @field = field
        end
      end
    end
  end
end

File: crates/agrr-domain/src/field/results/field_with_farm.rs
use crate::field::entities::FieldEntity;
use crate::field::results::FarmRecord;

/// Ruby: `Domain::Field::Results::FieldWithFarm`
#[derive(Debug, Clone, PartialEq)]
pub struct FieldWithFarm {
    pub farm: FarmRecord,
    pub field: FieldEntity,
}

impl FieldWithFarm {
    pub fn new(farm: FarmRecord, field: FieldEntity) -> Self {
        Self { farm, field }
    }
}
